Py4GWCoreLib/Py4GWcorelib.py

```diff
@@ -857,13 +911,21 @@ def add_action(self, action, *args, **kwargs):
         self.queue.append((action, args, kwargs))
         self._step_ids.append(self._step_counter)  # Track step ID
         self._step_counter += 1
-        
+
     def execute_next(self):
         if self.queue:
             action, args, kwargs = self.queue.popleft()
             self._last_step_id = self._step_ids.popleft()  # Extract step ID
+
+            account_name = PyPlayer.PyPlayer().account_name
+            logger = self._get_logger_for_account_name(account_name)
+
+            if logger:
+                logger.info(f"Called {action.__name__}(*{args}, **{kwargs})")
+
```

Comment:
@apoguita I'd rather log before we call the action, and not in the history, because if it fails at the call, we don't see the history.